src/server/zset_family.cc

```diff
@@ -1979,17 +2000,35 @@ void ZRangeGeneric(CmdArgList args, Transaction* tx, SinkReplyBuilder* builder,
 }
 
 void ZRankGeneric(CmdArgList args, bool reverse, Transaction* tx, SinkReplyBuilder* builder) {
+  if (args.size() > 3) {
```

Comment:
would you like to try replacing this parsing logic with `facade::CmdArgParser` ?
we usually replace the old parsing code with this as it's more clear this way.

Reply:
It turns out that redis does this check first. so i left this check in favor to throw the same error messages.

Providing this check by using the parser isn't obvious. 
I tried to write a clean code by using parser for that and failed =D